src/main/java/dev/jbang/cli/Jdk.java

```diff
@@ -235,6 +236,37 @@ public Integer javaEnv(
 		return EXIT_OK;
 	}
 
+	@CommandLine.Command(name = "exec", aliases = "x", description = "Executes the given command using the default (or specified) JDK.")
+	public Integer exec(
+			@CommandLine.Option(names = { "-j",
+					"--java" }, description = "JDK version to use for executing the command.") String versionOrId,
+			@CommandLine.Parameters(index = "0..*", arity = "1..*", description = "Command to execute") List<String> args)
+			throws IOException, InterruptedException {
+		JdkManager jdkMan = jdkProvidersMixin.getJdkManager();
+		dev.jbang.devkitman.Jdk jdk = null;
+		if (versionOrId != null && JavaUtil.isRequestedVersion(versionOrId)) {
+			jdk = jdkMan.getJdk(versionOrId, JdkProvider.Predicates.canUpdate);
+		}
+		if (jdk == null || !jdk.isInstalled()) {
+			jdk = jdkMan.getOrInstallJdk(versionOrId);
+		}
+		if (jdk.isInstalled()) {
+			Path home = ((dev.jbang.devkitman.Jdk.InstalledJdk) jdk).home();
+			String path = home + java.io.File.separator + "bin" + java.io.File.pathSeparator + System.getenv("PATH");
+			Path cmd = Util.searchPath(args.get(0), path);
+			if (cmd != null) {
+				args.set(0, cmd.toString());
+			}
+			ProcessBuilder pb = CommandBuffer.of(args).asProcessBuilder();
```

Comment:
This doesn't seem right. 

Shouldn't jbang be printing out the command for the surrounding shell script to execute ?

Reply:
after seeing your rationale for the issue I grok it.

Thought initially it was just to run a java command ...not about running with env setup.